xpk.py

```diff
@@ -3862,12 +3870,43 @@ def get_pathways_rm_args(args) -> str:
               - --pathways_persistent_compilation_cache=false
               - --pathways_compilation_mode=compile_at_worker
               - --pathways_tmp_dir_pattern={args.pathways_gcs_location}
-              - --pathways_resource_manager_expected_num_worker_jobs={args.num_slices}"""
+              - --pathways_expected_instances={expected_instances}"""
   if args.use_pathways:
-    return yaml.format(args=args)
+    return yaml.format(args=args, expected_instances=compute_pathways_expected_instances(args, system))
   else:
     return ""
 
+def compute_pathways_expected_instances(args, system: SystemCharacteristics) -> str:
```

Comment:
Can we add some tests for these new functions? Like testing at least one valid and one invalid accelerator type.